### PR TITLE
ref(store): Random Kafka partitioning for sessions

### PR DESCRIPTION
**Pull request by jan-auer**
Instead of using the session ID as partition key for producing sessions, assign
a random partitioning key. This ensures more even distribution of sessions on
the Kafka topic, even if clients are sending many updates for the same session.

We have noticed intermittent imbalance on the partitions. The theory is that
clients are sending repeat updates for the same session (ID) when a high number
of errors are occurring within that session. Every session update would go to
the same partition unless we use random partitioning.

Snuba does not have logical constraints on the partitioning.

### CODE REVIEW

File: relay-server/src/actors/store.rs [hunk 1]
```diff
@@ -673,8 +673,8 @@ impl KafkaMessage {
             Self::Attachment(message) => message.event_id.0,
             Self::AttachmentChunk(message) => message.event_id.0,
             Self::UserReport(message) => message.event_id.0,
-            Self::Session(message) => message.session_id,
-            Self::Metric(_message) => Uuid::nil(), // TODO(ja): Determine a partitioning key
+            Self::Session(_message) => Uuid::nil(), // Explicit random partitioning for sessions
```
Comment (jan-auer) [inline relay-server/src/actors/store.rs:676]: Note that this NIL is overwritten immediately after the match block.

**Review by jjbayer — APPROVED** [review]
As long as we are sure that the partitioning by ID wasn't there for a reason, this looks good to me!

**Comment by jan-auer** [comment]
Asking for @fpacifici's blessing before merging this :)

File: relay-server/src/actors/store.rs (hunk 1 above)
Comment (untitaker) [reply relay-server/src/actors/store.rs:676]: nit, can we note down the observed scenario that motivated this change?

**Review by untitaker — APPROVED** [review]

File: relay-server/src/actors/store.rs (hunk 1 above)
Comment (jan-auer) [reply relay-server/src/actors/store.rs:676]: I was hoping in code it's enough to say that we explicitly want uniform distribution. The full reasoning is in the PR description, and given that it's a hypothesis I wasn't gonna put this in the code.

**Pull request merged**